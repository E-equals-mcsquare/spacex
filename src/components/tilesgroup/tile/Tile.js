import React from 'react';
import styles from './Tile.module.css'

function Tile(props) {
    return (
      <div className={styles.tile}>
          <img className={styles.flightlogo} src={props.flightimageurl} alt={props.missioname}/>
          <div className={styles.missiontitle}>{props.missionname}</div>
          <div className={styles.kpititle}>Mission Ids:</div>
          {props.missionids.map((obj) => {
              return (
                <div className={styles.kpivalue}>&#8226; {obj}</div>
              )
          })}
          <div className={styles.kpititle}>Launch Year: <span className={styles.kpivalue}>{props.launchyear}</span></div>
          <div className={styles.kpititle}>Successful Launch: <span className={styles.kpivalue}>{props.successfullaunch}</span></div>
          <div className={styles.kpititle}>Successful Landing: <span className={styles.kpivalue}>{props.successfulland}</span></div>
      </div>
    );
  }
  
export default Tile;