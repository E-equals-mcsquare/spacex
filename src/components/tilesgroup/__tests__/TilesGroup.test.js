import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Tile  from '../tile/Tile'

describe('Tile', () => {
  it('renders Tile', () => {
    const wrapper = shallow(<Tile 
        flightimageurl="https://sampleimage.com"
        missionids={["ABC", "DEF"]}
        missionname="Mission 1"
        launchyear="2014"
        successfullaunch="true"
        successfulland="false" />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})