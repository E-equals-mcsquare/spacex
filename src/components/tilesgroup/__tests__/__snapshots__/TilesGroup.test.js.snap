// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders Tile 1`] = `
<div
  className="tile"
>
  <img
    className="flightlogo"
    src="https://sampleimage.com"
  />
  <div
    className="missiontitle"
  >
    Mission 1
  </div>
  <div
    className="kpititle"
  >
    Mission Ids:
  </div>
  <div
    className="kpivalue"
    key="ABC"
  >
    • 
    ABC
  </div>
  <div
    className="kpivalue"
    key="DEF"
  >
    • 
    DEF
  </div>
  <div
    className="kpititle"
  >
    Launch Year: 
    <span
      className="kpivalue"
    >
      2014
    </span>
  </div>
  <div
    className="kpititle"
  >
    Successful Launch: 
    <span
      className="kpivalue"
    >
      true
    </span>
  </div>
  <div
    className="kpititle"
  >
    Successful Landing: 
    <span
      className="kpivalue"
    >
      false
    </span>
  </div>
</div>
`;
